File: code/backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/taxe38/Locale38DetailAdminService.java
package ma.zs.univ.service.facade.admin.taxe38;

import java.util.List;
import ma.zs.univ.bean.core.taxe38.Locale38Detail;
import ma.zs.univ.dao.criteria.core.taxe38.Locale38DetailCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface Locale38DetailAdminService extends  IService<Locale38Detail,Locale38DetailCriteria>  {

    List<Locale38Detail> findByTypeLocale38DetailId(Long id);
    int deleteByTypeLocale38DetailId(Long id);
    long countByTypeLocale38DetailCode(String code);




}

File: code/frontend/src/app/shared/service/admin/taxe38/Locale38DetailAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {Locale38DetailDto} from 'src/app/shared/model/taxe38/Locale38Detail.model';
import {Locale38DetailCriteria} from 'src/app/shared/criteria/taxe38/Locale38DetailCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class Locale38DetailAdminService extends AbstractService<Locale38DetailDto, Locale38DetailCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): Locale38DetailDto {
        return new Locale38DetailDto();
    }

    public constrcutCriteria(): Locale38DetailCriteria {
        return new Locale38DetailCriteria();
    }

    get API() {
        return environment.apiUrlUnivservice + 'admin/locale38Detail/';
    }
}
